### DISCUSSION TITLE
Replacing content of a notebook with json string in a custom extension

### DISCUSSION THREAD

**Discussion by eugene-f02** [jupyterlite/jupyterlite#862] [Q&A] [answered]
Hi
I'm working on a custom extension and I'm looking for a way to programmatically populate the notebook's content with a JSON string. 

I discovered this method: `DocumentRegistry.IModel.fromString(value: string): void` which does the job.

I have a notebook widget, so I simply call ` notebook.context.model.fromString(content)`, and it does indeed replaces the content of the notebook with my `content` JSON string.

However, I'm noticing this error in the console which is also triggered whenever I try to save anything to the notebook that has been populated with JSON string using `fromString` method:

<img width="1007" alt="Screen Shot 2022-11-04 at 2 34 35 AM" src="https://user-images.githubusercontent.com/57968389/199907304-c2e5ddd9-3800-47e4-90c9-ee1da0be6ded.png">

The error mentions something about `contentChanged`. After looking through the docs of  `DocumentRegistry.IModel` I discovered that it mentions the following in the description for `fromString` method: `Should emit a [contentChanged] signal.`

I'm wondering what it means and whether I'm missing something when I simply call the `fromString` method. Should there be something else I should be calling before/after `fromString` method in order to avoid the error? How exactly can I emit this signal?

Thanks in advance!

**Comment by bollwyvl** [comment, marked as answer]
Yeah, have been seeing that with the celltoolbar recently. If you don't mind losing the feature, you can try this in `jupyter-lite.json`:

```json
{
  "jupyter-lite-schema-version": 0,
  "jupyter-config-data": {
    "disabledExtensions": [
      "@jupyterlab/cell-toolbar-extension:plugin"
    ]
  }
}

```

**Reply by eugene-f02** [reply]
Thanks, @bollwyvl! That worked
